Lib/test/test_asyncio/test_tasks.py

```diff
@@ -2661,5 +2656,42 @@ def coro():
         self.assertEqual(result, 11)
 
 
+class CompatibilityTests(test_utils.TestCase):
+    # Tests for checking a bridge between old-styled coroutines
+    # and async/await syntax
+
+    def setUp(self):
+        super().setUp()
+        self.loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(None)
+
+    def tearDown(self):
+        self.loop.close()
+        self.loop = None
+        super().tearDown()
+
+    def test_yield_from_awaitable(self):
+
+        @asyncio.coroutine
+        def coro():
+            yield from asyncio.sleep(0, loop=self.loop)
+            return 'ok'
+
+        result = self.loop.run_until_complete(coro())
+        self.assertEqual('ok', result)
+
+    def test_await_old_style_coro(self):
+
+        @asyncio.coroutine
+        def coro():
+            return 'ok'
```

Comment:
Please test both types of old-style coroutines in this test: the ones that have 'yield from' in them and the ones that don't:

```python
@coroutine
def coro1():
    return 'ok'

@coroutine
def coro2():
    yield from asyncio.sleep(0)
    return 'ok'
```